arrow-row/src/variable.rs

```diff
@@ -243,6 +244,88 @@ pub fn decode_binary<I: OffsetSizeTrait>(
     unsafe { GenericBinaryArray::from(builder.build_unchecked()) }
 }
 
+fn decode_binary_view_inner(
+    rows: &mut [&[u8]],
+    options: SortOptions,
+    check_utf8: bool,
+) -> BinaryViewArray {
+    let len = rows.len();
+
+    let mut null_count = 0;
+
+    let nulls = MutableBuffer::collect_bool(len, |x| {
+        let valid = rows[x][0] != null_sentinel(options);
+        null_count += !valid as usize;
+        valid
+    });
+
+    let values_capacity: usize = rows.iter().map(|row| decoded_len(row, options)).sum();
+    let mut values = MutableBuffer::new(values_capacity);
```

Comment:
We reserved the max amount of bytes for this buffer. But techinically, the buffer can only contain strings that are longer than 12 bytes.

This means that we are slightly less memory efficient. But the advantage is that we can easily do utf8 validation later.

Reply:
I think it would be valuable to add that context as a comment here as well 

```suggestion
    // Reserve the max amount of bytes possible for this buffer.
    // However, only strings longer than 12 bytes are copied here
    // This means it is less memory efficient but faster to do utf8 validation 
    let mut values = MutableBuffer::new(values_capacity);
```